File: xpilot/contrib/jxpmap/python/MapCheckPoint.py
from MapObject import MapObject


class MapCheckPoint(MapObject):
    def __init__(self, x = 0, y = 0, bmstore = None):
        MapObject.__init__(self, x, y, 35 * 64, 35 * 64,
                           bmstore["checkpoint.gif"])

    def printXml(self, file):
        b = self.getBounds()
        file.write('<Check x="%d" y="%d"/>\n' % (b.x + b.width / 2,
                                               b.y + b.height / 2))

File: xpilot/contrib/jxpmap/org/xpilot/jxpmap/MapCheckPoint.java
package org.xpilot.jxpmap;

import java.io.PrintWriter;
import java.io.IOException;

public class MapCheckPoint extends MapObject {


    public MapCheckPoint () {
        this(0, 0);
    }


    public MapCheckPoint (int x, int y) {
        super("checkpoint.gif", x, y, 35 * 64, 35 * 64);
    }


    public void printXml (PrintWriter out) throws IOException {
        out.print("<Check x=\"");
        out.print(getBounds().x + getBounds().width / 2);
        out.print("\" y=\"");
        out.print(getBounds().y + getBounds().height / 2);
        out.println("\"/>");
    }
}
